### DISCUSSION TITLE
tasmota32-lvgl.bin for esp32-S3-WROOM-1?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#20283] [Q&A]
i installed this [binary](http://ota.tasmota.com/tasmota32/release/tasmota32-lvgl.bin) on this [board](https://www.aliexpress.com/item/1005005099968475.html?spm=a2g0o.order_list.order_list_main.28.21ef1802k9ywQK).

It has a ESP32-S3-WROOM-1.

i got those Output after installing and opened the console:
```
...
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 0. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 0. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Invalid chip id. Expected 9 read 0. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
...
```

**Comment by ChrisVonFrosta** [comment]
Develpment binary start working

**Reply by sfromis** [reply]
You need specific binaries for each ESP32 variant ESP32, ESP32-S2, ESP32-S6, ESP32-C3, ESP32-C2, ESP32-C6 and ESP32-Solo1, meaning that for ESP32-S3, your selection should be among binaries with names starting with `tasmota32s3`.

And when you do initial wired flashing, you should select among binaries with names ending with `.factory.bin`, at the correct offset of 0.

To make a workable selection, using the Tasmota web installer https://tasmota.github.io/install/ is much recommended, where you can select a LVGL binary. That would behind the curtains become `tasmota32s3-lvgl.factory.bin` which you can also find at https://github.com/Jason2866/Tasmota-specials/tree/firmware/firmware/tasmota32/other

**Comment by ChrisVonFrosta** [comment]
Tasmota ran with the developer binary (the display not yet!). So i tried the '.factory.bin' as you said.
than i got the same error output like above.

```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
```

**Comment by ChrisVonFrosta** [comment]
i downloaded this [file](https://github.com/Jason2866/Tasmota-specials/blob/firmware/firmware/tasmota32/other/tasmota32s3-lvgl.factory.bin)

**Reply by sfromis** [reply]
You did not tell **exactly** how you flashed, meaning all details, like the full command if you used `esptool.py`.

Note that one risk factor is also if you need a binary supporting a specific type of communication with the flash/PSRAM chips, like QIO or OPI. However, better get the basics clear before assuming that.

While I'm not 100% that I found the right data, I did find a suggestion that 
`Invalid chip id. Expected 9 read 5. Bootloader for wrong chip?`
would be ESP32-C3 for chip id 5, which indeed would be a wrong binary.

**Comment by ChrisVonFrosta** [comment]
i flashed with chrome -> WebInstaller.
both files, [tasmota32s3-lvgl.factory.bin](https://github.com/Jason2866/Tasmota-specials/blob/firmware/firmware/tasmota32/other/tasmota32s3-lvgl.factory.bin) and [tasmota32s3-lvgl.bin](https://github.com/Jason2866/Tasmota-specials/blob/firmware/firmware/tasmota32/other/tasmota32s3-lvgl.bin) result in "wrog chip?"

tasmota starts runing with the binary `Tasmote32 LVGL (english)` from the WebInstaller's Dropdown Menu under `Development`

**Reply by sfromis** [reply]
Not clear what you specifically meant by that line with "both files".

`Tasmota32 LVGL (english)` is not a specific binary, but a selection which will result in a binary being selected to match the architecture.

With another ESP32-S3, in the Information page, I'm seeing this indicator of the chip:
`ESP Chip Id	11081760 (ESP32-S3 v0.1)`

If you have something like that, you should be able to upgrade to `tasmota32s3-lvgl.bin` (this binary is only for OTA upgrades, not for initial wired flashing).
